pkg/metrics/queue.go

```diff
@@ -146,6 +150,7 @@ func (m *QueueMetrics) Reset() {
 	m.appMetricsSubsystem.Reset()
 	m.resourceMetricsLabel.Reset()
 	m.resourceMetricsSubsystem.Reset()
+	m.knownResourceTypes = make(map[string]struct{})
```

Comment:
Guard this with a lock (see previous comment).

Reply:
Addressed in latest PR, thanks!